include/ada/url_base.h

```diff
@@ -14,7 +14,9 @@
 namespace ada {
 
   /**
-   * @private
+   * A url_base contains a few attributes: is_valid, has_opaque_path and type.
+   * All non-trivial implementation details are in derived classes such as
+   * ada::url and ada::url_aggregator.
```

Comment:
> Do we want to expose url_base in documentation? 

Users would not hold a url_base instance since it is an abstract class, but it does not seem harmful that folks see it in the documentation.